jerry-ext/include/jerryscript-ext/section.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRY_EXT_SECTION_H
+#define JERRY_EXT_SECTION_H
+
+/**
+ * Define the name of a section
+ *
+ * @param name the name of the section without quotes
+ */
+#ifdef __MACH__
+#define JERRYX_SECTION_NAME(name) "__DATA," #name
+#else /* !__MACH__ */
+#define JERRYX_SECTION_NAME(name) #name
+#endif /* __MACH__ */
+
+#define JERRYX_SECTION_APPEND(name) \
+__attribute__ ((used, section (JERRYX_SECTION_NAME (name)), aligned (sizeof (void *))))
+
+/**
+ * Declare references to a section that contains an array of items.
+ *
+ * @param name the name of the section (without quotes)
+ * @param type the type of the elements stored in the array
+ *
+ * This macro needs to be placed before any occurrence of JERRYX_SECTION_ITERATE, for a given section.
+ */
+#ifdef __MACH__
+#define JERRYX_SECTION_DECLARE(name, type)                                 \
+extern const type __start_ ## name[] __asm("section$start$__DATA$" #name); \
+extern const type __stop_ ## name[] __asm("section$end$__DATA$" #name);
```

Comment:
Actually, changing it to a pointer doesn't work, because the `__stop_ ## name` becomes null, and it tries to allocate a pile of memory. I guess pointers and arrays are not the same to the linker.